cognite/client/data_classes/annotation_types/images.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from cognite.client.data_classes.annotation_types.primitives import (
+    BoundingBox,
+    CdfResourceRef,
+    Polygon,
+    PolyLine,
+    VisionResource,
+)
+
+
+@dataclass
+class ObjectDetection(VisionResource):
+    label: str
+    confidence: Optional[float]
+    # A valid object detection instance needs to have *exactly one* of these
+    bounding_box: Optional[BoundingBox] = None
+    polygon: Optional[Polygon] = None
+    polyline: Optional[PolyLine] = None
+
+    def __post_init__(self) -> None:
+        if isinstance(self.bounding_box, Dict):
+            self.bounding_box = BoundingBox(**self.bounding_box)
+        if isinstance(self.polygon, Dict):
+            self.polygon = Polygon(self.polygon)
```

Comment:
I think this could be a bug that we haven't caught yet:
```suggestion
            self.polygon = Polygon(**self.polygon)
```

Reply:
[99746be](https://github.com/cognitedata/cognite-sdk-python/pull/1014/commits/99746bed0cfd8ab17858c4b335f460db0b9a14c4)